src/credentials/GroupDataProvider.h

```diff
@@ -143,7 +143,7 @@ class GroupDataProvider
         {}
 
         // The actual keys for the group key set
-        EpochKey epoch_keys[kEpochKeysMax];
+        EpochKey epoch_keys[kEpochKeysMax] = {};
```

Comment:
I am curious how we ended up in a situation where `num_keys_used` was nonzero but some of the "used" keys were not initialized.  What was the actual warning here?  What location?  Do we perhaps have a callstack?

Reply:
This led me to an uninitialised start_time variable in EpochKey, for the temporary key "temporary ipk 01". Maybe initialising only the start_time variable would be a more readable change. I have added such a change now. Here a piece of log valgrind:

```
==587390== 3 errors in context 8 of 11:
==587390== Conditional jump or move depends on uninitialised value(s)
==587390==    at 0x484ED28: strlen (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==587390==    by 0x13F311: std::char_traits<char>::length(char const*) (char_traits.h:399)
==587390==    by 0x141DAD: std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string<std::allocator<char> >(char const*, std::allocator<char> const&) (basic_string.h:536)
==587390==    by 0x435729: chip::DeviceLayer::Internal::ChipLinuxStorageIni::AddEntry(char const*, char const*) (CHIPLinuxStorageIni.cpp:363)
==587390==    by 0x432B10: chip::DeviceLayer::Internal::ChipLinuxStorage::WriteValueStr(char const*, char const*) (CHIPLinuxStorage.cpp:221)
==587390==    by 0x432D80: chip::DeviceLayer::Internal::ChipLinuxStorage::WriteValueBin(char const*, unsigned char const*, unsigned long) (CHIPLinuxStorage.cpp:267)
==587390==    by 0x44CDC3: chip::DeviceLayer::PersistedStorage::KeyValueStoreManagerImpl::_Put(char const*, void const*, unsigned long) (KeyValueStoreManagerImpl.cpp:83)
==587390==    by 0x4635E9: chip::DeviceLayer::PersistedStorage::KeyValueStoreManager::Put(char const*, void const*, unsigned long) (KeyValueStoreManager.h:222)
==587390==    by 0x4DABEF: chip::KvsPersistentStorageDelegate::SyncSetKeyValue(char const*, void const*, unsigned short) (KvsPersistentStorageDelegate.h:97)
==587390==    by 0x1B9CBB: chip::PersistentData<128ul>::Save(chip::PersistentStorageDelegate*) (PersistentData.h:55)
==587390==    by 0x4B0594: chip::Credentials::GroupDataProviderImpl::SetKeySet(unsigned char, chip::Span<unsigned char const> const&, chip::Credentials::GroupDataProvider::KeySet const&) (GroupDataProviderImpl.cpp:1556)
==587390==    by 0x199DE2: emberAfOperationalCredentialsClusterAddNOCCallback(chip::app::CommandHandler*, chip::app::ConcreteCommandPath const&, chip::app::Clusters::OperationalCredentials::Commands::AddNOC::DecodableType const&) (operational-credentials-server.cpp:665)
==587390==  Uninitialised value was created by a stack allocation
==587390==    at 0x19982D: emberAfOperationalCredentialsClusterAddNOCCallback(chip::app::CommandHandler*, chip::app::ConcreteCommandPath const&, 
```

Reply:
@DamMicSzm Thank you, that stack is very helpful!  The code there is:

```
    keyset.num_keys_used = 1;
    memcpy(keyset.epoch_keys[0].key, ipkValue.data(), Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES);
```
and it in fact does not set `keyset.epoch_keys[0].start_time`, which is clearly wrong.  Compare the `SetSingleIpkEpochKey` function, which is doing something similar but does:

```
    ipkKeySet.epoch_keys[0].start_time = 0;
    memcpy(&ipkKeySet.epoch_keys[0].key, ipk_epoch_span.data(), ipk_epoch_span.size());
```

So the right fix is to fix `emberAfOperationalCredentialsClusterAddNOCCallback` to set `start_time` to 0.

Reply:
Thank you for your help in fixing this problem. I hope now it will be fine.